services/src/main/java/org/keycloak/email/DefaultEmailSenderProvider.java

```diff
@@ -185,22 +185,16 @@ protected String retrieveEmailAddress(UserModel user) {
     }
 
     private void setupTruststore(Properties props) {
-        boolean checkServerIdentity = true;
-
         JSSETruststoreConfigurator configurator = new JSSETruststoreConfigurator(session);
 
         SSLSocketFactory factory = configurator.getSSLSocketFactory();
         if (factory != null) {
             props.put("mail.smtp.ssl.socketFactory", factory);
             if (configurator.getProvider().getPolicy() == HostnameVerificationPolicy.ANY) {
                 props.setProperty("mail.smtp.ssl.trust", "*");
-                checkServerIdentity = false;
+                props.put("mail.smtp.ssl.checkserveridentity", Boolean.FALSE.toString());
             }
         }
-
-        if (checkServerIdentity) {
```

Comment:
Minor: I would maybe keep the `checkServerIdentity` variable and use here something like `props.put("mail.smtp.ssl.checkserveridentity", String.valueOf(checkServerIdentity);`.

Just in case that "default" changes again in the future, we would have it always overriden based on our settings. But that is minor nitpick and doesn't worth blocking this PR, so approving anyway.

Reply:
@mposolda I thought about it, always setting the variable with true or false, but then I decided that this is not probably going to change again. Setting this defaulted to true is the logical option based on secure by default current approach.